object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/ObjectConstructionAnnotatedTypeFactory.java

```diff
@@ -108,6 +98,12 @@ public ObjectConstructionAnnotatedTypeFactory(final BaseTypeChecker checker) {
     super(checker);
     TOP = AnnotationBuilder.fromClass(elements, CalledMethodsTop.class);
     BOTTOM = AnnotationBuilder.fromClass(elements, CalledMethodsBottom.class);
+
+    // for framework support
+    frameworkSupports = new ArrayList<FrameworkSupport>();
+    frameworkSupports.add(new AutoValueSupport(this));
```

Comment:
We should make these configurable, using a similar mechanism to the useValueChecker flag immediately below. I think the current default is correct, but it should be possible to turn them off. Perhaps if no flag is specified, it uses Lombok and AutoValue. But, there is a flag `-AframeworksSupported=` that can take either a comma-separated list or `"none"`. So, for example:

`-AframeworksSupported=none` results in no frameworks,
`-AframeworksSupported=lombok` results in only lombok, and
`-AframeworksSupported=lombok,autovalue` results in both (that is, it's equivalent to passing no `-AframeworksSupported` argument).

Reply:
@kelloggm the tricky thing here is threading the settings into the returns receiver checker as well.  Maybe we can add the flags in a follow-up PR?

Reply:
Sure, that's fine with me.